test/custom-undici-instance.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const t = require('tap')
+const Fastify = require('fastify')
+const undici = require('undici')
+const http = require('http')
+const get = require('simple-get').concat
+const From = require('..')
+
+const target = http.createServer((req, res) => {
+  t.pass('request proxied')
+  t.equal(req.method, 'GET')
+  t.equal(req.url, '/')
+  t.equal(req.headers.connection, 'keep-alive')
+  res.statusCode = 205
+  res.setHeader('Content-Type', 'text/plain')
+  res.setHeader('x-my-header', 'hello!')
+  res.end('hello world')
+})
+
+t.test('use a custom instance of \'undici\'', async t => {
+  t.teardown(target.close.bind(target))
+
+  await new Promise((resolve, reject) => target.listen(0, err => err ? reject(err) : resolve()))
+
+  t.test('custom Pool', t => {
+    const instance = Fastify()
+    t.teardown(instance.close.bind(instance))
+    instance.register(From, {
+      base: `http://localhost:${target.address().port}`,
```

Comment:
Unless undici adds a way to get the target url, the base has two be specified in the options. Is that ok?

Reply:
It's ok for now, however we might want to change it in undici! Maybe a PR there?

Reply:
Yes, that makes sense 😉